ccl/importccl: TestCSVImportCanBeResumed failed

[(ccl/importccl).TestCSVImportCanBeResumed failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2707940&tab=buildLog) on [master@2572200f7612c6508a52735a6a18767cfb7cc09d](https://github.com/cockroachdb/cockroach/commits/2572200f7612c6508a52735a6a18767cfb7cc09d):

```
  github.com/cockroachdb/cockroach/pkg/ccl/importccl.(*readImportDataProcessor).Start()
      /go/src/github.com/cockroachdb/cockroach/pkg/ccl/importccl/import_processor.go:105 +0xa4
  github.com/cockroachdb/cockroach/pkg/sql/execinfra.(*ProcessorBase).Run()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/execinfra/processorsbase.go:764 +0x8e
  github.com/cockroachdb/cockroach/pkg/ccl/importccl.(*readImportDataProcessor).Run()
      <autogenerated>:1 +0x57
  github.com/cockroachdb/cockroach/pkg/sql/flowinfra.(*FlowBase).Run()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/flowinfra/flow.go:382 +0x2c6
  github.com/cockroachdb/cockroach/pkg/sql/rowflow.(*rowBasedFlow).Run()
      <autogenerated>:1 +0x84
  github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).Run()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/distsql_running.go:367 +0x7a7
  github.com/cockroachdb/cockroach/pkg/sql.DistIngest.func6()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/distsql_plan_csv.go:332 +0x1b2
  github.com/cockroachdb/cockroach/pkg/util/ctxgroup.Group.GoCtx.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/ctxgroup/ctxgroup.go:166 +0x4e
  golang.org/x/sync/errgroup.(*Group).Go.func1()
      /go/src/github.com/cockroachdb/cockroach/vendor/golang.org/x/sync/errgroup/errgroup.go:57 +0x94

Goroutine 26 (running) created at:
  golang.org/x/sync/errgroup.(*Group).Go()
      /go/src/github.com/cockroachdb/cockroach/vendor/golang.org/x/sync/errgroup/errgroup.go:54 +0x73
  github.com/cockroachdb/cockroach/pkg/util/ctxgroup.Group.GoCtx()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/ctxgroup/ctxgroup.go:165 +0xe4
  github.com/cockroachdb/cockroach/pkg/sql.DistIngest()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/distsql_plan_csv.go:328 +0x128d
  github.com/cockroachdb/cockroach/pkg/ccl/importccl.(*importResumer).Resume()
      /go/src/github.com/cockroachdb/cockroach/pkg/ccl/importccl/import_stmt.go:1632 +0xe38
  github.com/cockroachdb/cockroach/pkg/ccl/importccl.(*cancellableImportResumer).Resume()
      /go/src/github.com/cockroachdb/cockroach/pkg/ccl/importccl/import_processor_test.go:554 +0x1cc
  github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).stepThroughStateMachine.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/jobs/registry.go:1170 +0x141
  github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).stepThroughStateMachine()
      /go/src/github.com/cockroachdb/cockroach/pkg/jobs/registry.go:1171 +0xa85
  github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).runJob()
      /go/src/github.com/cockroachdb/cockroach/pkg/jobs/adopt.go:250 +0x43e
  github.com/cockroachdb/cockroach/pkg/jobs.(*StartableJob).Start.func3()
      /go/src/github.com/cockroachdb/cockroach/pkg/jobs/jobs.go:867 +0x25c
  github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTask.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:351 +0x149
==================
E210223 06:16:33.560091 1505 jobs/adopt.go:255  [n1] 1  job 635668376058003457: adoption completed with error job 635668376058003457: could not mark as reverting: context canceled: job with status pause-requested cannot be reverted
    import_processor_test.go:733: Resume pos: 30
E210223 06:16:34.408378 635 jobs/registry.go:672  [-] 2  error claiming jobs: could not query jobs table: claim-jobs: context canceled
E210223 06:16:34.408913 633 jobs/registry.go:662  [-] 3  failed to serve pause and cancel requests: could not query jobs table: cancel/pause-requested: context canceled
E210223 06:16:34.410822 635 jobs/registry.go:682  [-] 4  error processing claimed jobs: could not query for claimed jobs: select-running/get-claimed-jobs: context canceled
    import_processor_test.go:747: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestCSVImportCanBeResumed385659822
    testing.go:1038: race detected during execution of test
--- FAIL: TestCSVImportCanBeResumed (1.85s)

```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-race -parallel=4

```
make stressrace TESTS=TestCSVImportCanBeResumed PKG=./pkg/ccl/importccl TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestCSVImportCanBeResumed.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

--- a/pkg/ccl/backupccl/backup_processor.go
+++ b/pkg/ccl/backupccl/backup_processor.go
@@ -109,17 +109,11 @@ func newBackupDataProcessor(
 
 // Start is part of the RowSource interface.
 func (bp *backupDataProcessor) Start(ctx context.Context) {
+	ctx = bp.StartInternal(ctx, backupProcessorName)
 	go func() {
 		defer close(bp.progCh)
 		bp.backupErr = runBackupProcessor(ctx, bp.flowCtx, &bp.spec, bp.progCh)
 	}()
-	ctx = bp.StartInternal(ctx, backupProcessorName)
-	// Go around "this value of ctx is never used" linter error. We do it this
-	// way instead of omitting the assignment to ctx above so that if in the
-	// future other initialization is added, the correct ctx is used.
-	// TODO(bulkio): check whether this context should be used in the closure
-	// above.
-	_ = ctx
 }
 
 // Next is part of the RowSource interface.

--- a/pkg/ccl/backupccl/split_and_scatter_processor.go
+++ b/pkg/ccl/backupccl/split_and_scatter_processor.go
@@ -203,6 +203,7 @@ func newSplitAndScatterProcessor(
 
 // Start is part of the RowSource interface.
 func (ssp *splitAndScatterProcessor) Start(ctx context.Context) {
+	ctx = ssp.StartInternal(ctx, splitAndScatterProcessorName)
 	go func() {
 		// Note that the loop over doneScatterCh in Next should prevent this
 		// goroutine from leaking when there are no errors. However, if that loop
@@ -213,13 +214,6 @@ func (ssp *splitAndScatterProcessor) Start(ctx context.Context) {
 		defer close(ssp.doneScatterCh)
 		ssp.scatterErr = ssp.runSplitAndScatter(scatterCtx, ssp.flowCtx, &ssp.spec, ssp.scatterer)
 	}()
-	ctx = ssp.StartInternal(ctx, splitAndScatterProcessorName)
-	// Go around "this value of ctx is never used" linter error. We do it this
-	// way instead of omitting the assignment to ctx above so that if in the
-	// future other initialization is added, the correct ctx is used.
-	// TODO(bulkio): check whether this context should be used in the closure
-	// above.
-	_ = ctx
 }
 
 type entryNode struct {

--- a/pkg/ccl/importccl/import_processor.go
+++ b/pkg/ccl/importccl/import_processor.go
@@ -100,6 +100,7 @@ func newReadImportDataProcessor(
 
 // Start is part of the RowSource interface.
 func (idp *readImportDataProcessor) Start(ctx context.Context) {
+	ctx = idp.StartInternal(ctx, readImportDataProcessorName)
 	// We don't have to worry about this go routine leaking because next we loop over progCh
 	// which is closed only after the go routine returns.
 	go func() {
@@ -107,13 +108,6 @@ func (idp *readImportDataProcessor) Start(ctx context.Context) {
 		idp.summary, idp.importErr = runImport(ctx, idp.flowCtx, &idp.spec, idp.progCh,
 			idp.seqChunkProvider)
 	}()
-	ctx = idp.StartInternal(ctx, readImportDataProcessorName)
-	// Go around "this value of ctx is never used" linter error. We do it this
-	// way instead of omitting the assignment to ctx above so that if in the
-	// future other initialization is added, the correct ctx is used.
-	// TODO(bulkio): check whether this context should be used in the closure
-	// above.
-	_ = ctx
 }
 
 // Next is part of the RowSource interface.

--- a/pkg/ccl/logictestccl/testdata/logic_test/multi_region_backup
+++ b/pkg/ccl/logictestccl/testdata/logic_test/multi_region_backup
@@ -1,5 +1,7 @@
 # LogicTest: multiregion-9node-3region-3azs
 
+skip flaky # see #60773
+
 query TTTT
 SHOW REGIONS
 ----
@@ -676,8 +678,6 @@ defaultdb
 postgres
 system
 test
-
-skip flaky # see #60773
 
 statement ok
 RESTORE DATABASE "mr-backup-1", "mr-backup-2" FROM 'nodelocal://1/mr-backup-combined/'

--- a/pkg/ccl/streamingccl/streamingest/stream_ingestion_processor.go
+++ b/pkg/ccl/streamingccl/streamingest/stream_ingestion_processor.go
@@ -157,15 +157,11 @@ func newStreamIngestionDataProcessor(
 // Start is part of the RowSource interface.
 func (sip *streamIngestionProcessor) Start(ctx context.Context) {
 	ctx = sip.StartInternal(ctx, streamIngestionProcessorName)
-	// Go around "this value of ctx is never used" linter error. We do it this
-	// way instead of omitting the assignment to ctx above so that if in the
-	// future other initialization is added, the correct ctx is used.
-	_ = ctx
 
 	evalCtx := sip.FlowCtx.EvalCtx
 	db := sip.FlowCtx.Cfg.DB
 	var err error
-	sip.batcher, err = bulk.MakeStreamSSTBatcher(sip.Ctx, db, evalCtx.Settings,
+	sip.batcher, err = bulk.MakeStreamSSTBatcher(ctx, db, evalCtx.Settings,
 		func() int64 { return storageccl.MaxImportBatchSize(evalCtx.Settings) })
 	if err != nil {
 		sip.MoveToDraining(errors.Wrap(err, "creating stream sst batcher"))
@@ -176,14 +172,14 @@ func (sip *streamIngestionProcessor) Start(ctx context.Context) {
 	startTime := timeutil.Unix(0 /* sec */, sip.spec.StartTime.WallTime)
 	eventChs := make(map[streamingccl.PartitionAddress]chan streamingccl.Event)
 	for _, partitionAddress := range sip.spec.PartitionAddresses {
-		eventCh, err := sip.client.ConsumePartition(sip.Ctx, partitionAddress, startTime)
+		eventCh, err := sip.client.ConsumePartition(ctx, partitionAddress, startTime)
 		if err != nil {
 			sip.MoveToDraining(errors.Wrapf(err, "consuming partition %v", partitionAddress))
 			return
 		}
 		eventChs[partitionAddress] = eventCh
 	}
-	sip.eventCh = sip.merge(sip.Ctx, eventChs)
+	sip.eventCh = sip.merge(ctx, eventChs)
 }
 
 // Next is part of the RowSource interface.

